src/index.js

```diff
@@ -230,7 +230,9 @@ class ExtractTextPlugin {
               chunk.removeModule(data.moduleToRemove);
               const deps = data.moduleToRemove.dependencies;
               deps.forEach((d) => {
-                chunk.removeModule(d.module);
+                if (d.module && d.module.loaders.length > 0) {
+                  chunk.removeModule(d.module);
+                }
```

Comment:
I've been thinking about the best way to handle CSS in dynamic chunks. What i've come up with is this: **`import()` needs to request both the js + the css.** 

I.e. 2 requests, rather than what I was doing before with `extract-css-chunks-webpack-plugin` where I made 2 js files (one with style-loader and another called `no_css.js` without style-loader). That adds considerably to the compilation time.

As far as HMR, this simplifies implementation--it's just one way to support.

The most important thing however is this: ***having a different setup in production than you have in development is a bad idea*** (i.e. where in development css is served in js via style-loader, and during production via external stylesheets). Having the same setup (external stylesheets) means going to production has less unforeseen bottlenecks. 

*If it works in development, it works in production.*

Developers new to Webpack expend a lot of wasted energy figuring this stuff out. I.e. fumbling around with file hashes, only embedding a stylesheet in production (whose file hash can be hard to find), etc. It's a frustrating stage in your project, especially when you thought you were done.

Since HMR works on stylesheets, it makes you think why was CSS ever injected from javascript? To save novice developers from pasting a `<link />` tag in addition  to `main.js`? Perhaps it's just legacy because that's how style loader was built first, or EWTP wasn't around in the beginning.

I think we gotta go back to square one. Maybe this is part of @sokra 's "masterplan." 

In short, it's a babel plugin that changes `import()` to `Promise.all([import(), fetchCss()])`. 

And now this plugin is easier than ever. It insures there is a matching CSS file per chunk, no matter whether it's named or dynamic. And it uses a very straightforward mechanism to reload CSS files corresponding to each module:

```
if (module.hot) {
	module.hot.accept();
	if (module.hot.data) {
		require(${loaderUtils.stringifyRequest(this, path.join(__dirname, "hotModuleReplacement.js"))})("${publicPath}", "%%extracted-file%%");
	}
}`;
```

Also, shouldn't compiling without ever moving css into js chunks be faster? Maybe this is part of the masterplan and needs to happen in core somewhere, but if CSS never makes its way into javascript modules/chunks, and is put off to the side early on, then we're talking not just easier for developers, but faster and maybe easier in implementation.